Automattic/jetpack
# Instant Search: Add default sort customizer control

Split out from #15727. 

#### Changes proposed in this Pull Request:
* Adds a Customizer control for setting the default sort order in the Jetpack Search overlay.
* Simplifies query string handling for sorting by using `sort` instead of a combination of `order` and `orderby`.
* Removes handling for `sort` selects created by Jetpack Search widgets. 
(A follow-up PR will remove these selects from being rendered in the first place. This can land after merging #15989 which includes handling for rendering instant search specific widgets.)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes, it adds the ability to define a default sort for the Jetpack Search overlay. This feature is only available for sites with a Jetpack Search product subscription. 

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Apply this change to a site with Jetpack Instant Search enabled. Make sure to add a search widget or Jetpack Search widget to your site's footer or sidebar in the customizer.

2. Navigate to the Jetpack Search section of the Customizer. 

3. Ensure that a new select has been added for changing the default sort for the search overlay.

> <img width="299" alt="Screen Shot 2020-05-29 at 4 57 44 PM" src="https://user-images.githubusercontent.com/4044428/83311845-b7aa9980-a1cd-11ea-8783-cad40cfa2b36.png">

4. Try selecting different default sort configurations. Ensure that the Customizer preview updates without a reload.

5. Publish your desired default sort configuration and return to your site.

6. Search your site using a search widget. Ensure that the search overlay appears with the default sort you chose in step 4. 

7. Profit!

#### Proposed changelog entry for your changes:
* Adds a Customizer control for setting the default sort order in the Jetpack Search overlay.
